lib/ffe.gi

```diff
@@ -600,7 +600,7 @@ InstallMethod( RootOfDefiningPolynomial,
 
     coeffs:= ShiftedCoeffs( coeffs[1], coeffs[2] );
     p:= Characteristic( F );
-    d:= ( Length( coeffs ) - 1 ) * DegreeOverPrimeField( F );
+    d:= DegreeOverPrimeField( F );
```

Comment:
Any ideas why this factor was there before?

Reply:
Good question. I dug a bit. That line was added by @ThomasBreuer in a commit from 1998-11-16 (hash 8a81e3cac9f5878891017aa3b6e9d060a0845f0d in the private git history). Looking at the code added there, a large part of it is identical to code in `FieldExtension`, including the bit computing `d` -- there, however, `F` was the base field, while here it is the extension field. So I suspect this was a copy&paste accident. Since this code was never triggered before I added the patches in this PR, we never noticed.

Reply:
One may in fact wonder whether it wouldn't be better to extract the common code into a new helper operation`RootOfPolynomial(F, poly)`. Then the above `RootOfDefiningPolynomial` method would essentially becomes `F -> RootOfPolynomial(PrimeField(F), DefiningPolynomial(F)`, and the `FieldExtension` method could also be shortened. And if somebody comes up with a better implementation than "try all field elements" to do this, they'd have to edit only one spot.